prog/vm/github_runner.rb

```diff
@@ -126,9 +126,22 @@ def before_run
   end
 
   label def start
+    hop_wait_concurrency_limit unless concurrency_available?
+    hop_allocate_vm
```

Comment:
We do the checking here and allocating in allocate_vm, that means multiple vms can pass this check and then surpass the limit in allocate_vm. To fix this, pick_vm needs to be in same transaction with concurrency_available?, which would require some shuffling with states unfortunately. Maybe we can find a simpler solution though, I did not think about it in depth.

Reply:
Discussed this offline, writing here for information sharing purposes. With the exclusive lock + the transactional states, this is not an issue. The total core calculation logic in the installation reads the usage directly from the strand excluding the first 2 states.